src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj

```diff
@@ -95,11 +104,18 @@
       (onWebSocketClose [statusCode ^String reason]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketClose statusCode reason))
+        (.countDown closureLatch)
         (on-close this statusCode reason))
       (onWebSocketBinary [^bytes payload offset len]
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketBinary payload offset len))
         (on-bytes this payload offset len))
+      (awaitClosure []
+        (try
+          (when-not (.await closureLatch 30 TimeUnit/SECONDS)
+            (log/debug (i18n/trs "Timed out awaiting closure of websocket from remote.")))
```

Comment:
I think this should be info level, and include a mention of the value of the timeout and any identifying information we have.

Reply:
Updated